### PR TITLE
readonly mode configurable for CVs

### PR DESCRIPTION
**Pull request by mandelsoft**
## Description

So far, only complete OCM repositories can be handled in ReadOnly mode.
THis PR introduces the possibility to set a dedicate CV access to readonly mode.
to avoid undesired writes to the backend.

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [x] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

- Related Issue # (issue)
- Closes # (issue)
- Fixes # (issue)
> Remove if not applicable

## Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help
- [ ] Separate ticket for tests # (issue/pr)

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Added to documentation?

- [ ] 📜 README.md
- [ ] 🙅 no documentation needed

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pkg/contexts/ocm/cpi/repocpi/backend.go [hunk 1]
```diff
@@ -29,7 +29,10 @@ type StorageBackendImpl interface {
 	io.Closer
 	GetContext() cpi.Context
 	GetSpecification() cpi.RepositorySpec
-	IsReadOnly() bool
+	// IsReadOnly returns readonly mode for repo and component,
+	// if key is given for the dedicated cv.
```
Comment (morri-son) [inline pkg/contexts/ocm/cpi/repocpi/backend.go:33]:
```suggestion
	// if key is given for the dedicated component version
```

File: cmds/ocm/topics/common/credentials/topic.go [hunk 2]
```diff
@@ -133,6 +133,16 @@ behaviours are described in the following list:
 
 ### Credential Providers
 
+Credential providers offer sets of named credentials from variuos sources,
+which might be directly mapped to consumer identities (if supported
+by the provider type).
+
+The type <code>Credentials</code> can be used to inline
+credentials for in credential configuration objects
```
Comment (morri-son) [inline cmds/ocm/topics/common/credentials/topic.go:141]:
```suggestion
credentials for credential configuration objects
```

File: docs/reference/ocm_credential-handling.md [hunk 3]
```diff
@@ -248,6 +248,16 @@ behaviours are described in the following list:
 
 ### Credential Providers
 
+Credential providers offer sets of named credentials from variuos sources,
+which might be directly mapped to consumer identities (if supported
+by the provider type).
+
+The type <code>Credentials</code> can be used to inline
+credentials for in credential configuration objects
```
Comment (morri-son) [inline docs/reference/ocm_credential-handling.md:256]:
```suggestion
credentials for credential configuration objects
```

**Review by morri-son — COMMENTED** [review]
lgtm beside the minor proposals I did

File: pkg/contexts/ocm/repositories/composition/repository.go [hunk 4]
```diff
@@ -178,7 +191,11 @@ func (v *VersionAccess) Close() error {
 }
 
 func (v *VersionAccess) IsReadOnly() bool {
-	return false
+	return v.readonly
+}
+
+func (v *VersionAccess) SetReadOnly() {
+	v.readonly = true
```
Comment (hilmarf) [inline pkg/contexts/ocm/repositories/composition/repository.go:198]: As discussed... this is now a one-way action!!
Once things are read-only, there is no way to revert and make it changeable again. From pure API perspective it looks a bit odd.

@mandelsoft - can you please document this and elaborate the details, why this is the right way to do?

**Review by hilmarf — COMMENTED** [review]
LGTM - but we should document somewhere and explain, why it's NOT possible to make things again write-able, once they are in read-only mode.

**Pull request merged**